qadence/backends/pyqtorch/convert_ops.py

```diff
@@ -214,13 +171,13 @@ def forward(
         self, state: torch.Tensor, values: dict[str, torch.Tensor] | None = None
     ) -> torch.Tensor:
         batch_size = state.size(-1)
-        return self.apply(self.matrices(values, batch_size), state)
+        return self.apply(self.unitary(values, batch_size), state)
 
     def apply(self, matrices: torch.Tensor, state: torch.Tensor) -> torch.Tensor:
         batch_size = state.size(-1)
         return _apply_batch_gate(state, matrices, self.qubits, self.n_qubits, batch_size)
 
-    def matrices(self, values: dict[str, torch.Tensor] | None, batch_size: int) -> torch.Tensor:
+    def unitary(self, values: dict[str, torch.Tensor] | None, batch_size: int) -> torch.Tensor:
         perm = (2, 0, 1)  # We permute the dims since torch.bmm expects the batch_dim at 0.
```

Comment:
Is it possible to construct values tensors such that the batch dim is the first dimension ? Just to avoid these implicit permutations.

Reply:
in native pyq , the batch dim is always last, its just to conform with this

Reply:
we do have to pass them back to pyq, so there is no way around it

Reply:
exactly

Reply:
its just to comply with the pyq state shape which has the batch dim last, if we do it mixed (for unitary matrices first and states last) we have to always permute and eerything becomes messy in pyq